cmd/vega/commands/node/node.go

```diff
@@ -442,14 +442,19 @@ func (n *Command) startBlockchainClients() error {
 		return nil
 	}
 
-	if n.conf.Blockchain.ChainProvider != blockchain.ProviderNullChain {
-		var err error
-		n.ethClient, err = ethclient.Dial(n.ctx, n.conf.Ethereum)
-		if err != nil {
-			return fmt.Errorf("could not instantiate ethereum client: %w", err)
-		}
-		n.ethConfirmations = ethclient.NewEthereumConfirmations(n.conf.Ethereum, n.ethClient, nil)
+	// We may not need ethereum client initialized when we have not
+	// provided the ethereum endpoint. We skip creating client here
+	// when RPCEnpoint is empty and the nullchain present.
+	if n.conf.Blockchain.ChainProvider == blockchain.ProviderNullChain && len(n.conf.Ethereum.RPCEndpoint) < 1 {
```

Comment:
there is no need to check for NullChain anymore:
```
n.conf.Blockchain.ChainProvider == blockchain.ProviderNullChain
```

Regardless of the provider, what determines the EthClient initialization is the `RPCEndpoint`. So you can remove NullChain check.